libraries/heft-config-file/src/ConfigurationFileBase.ts

```diff
@@ -690,8 +922,7 @@ export abstract class ConfigurationFileBase<TConfigurationFile, TExtraOptions ex
       }
 
       case PathResolutionMethod.resolvePathRelativeToProjectRoot: {
-        const packageRoot: string | undefined =
-          this._packageJsonLookup.tryGetPackageFolderFor(configurationFilePath);
+        const packageRoot: string | undefined = projectFolderPath;
```

Comment:
Shouldn't we be attempting to resolve this if not provided? Feels like this is a breaking change otherwise. If that's intended, then we should make it clear in the error message that it needs to be provided.

Reply:
Strictly speaking, this cannot be reliably computed (having subfolders contain package.json files is completely valid), so we should require it be passed in.

Reply:
That said, `ProjectConfigurationFile` does attempt to compute this value if it is not provided, before the loader is invoked.

Reply:
I see. Then I think that maybe the error message should be updated to more directly guide how to resolve the break? As it stands it's a bit opaque how one would fix this issue, ex. `No project folder path was provided but project root relative resolution was requested`